bk-api/bk_api.py

```diff
@@ -1316,102 +1253,95 @@ def get(self):
         Returns:
             dict: end-point id, private key, public key and certificate
         """
-        node_id = request.args.get("id", type=str)
+        logger.info("registration request: %r", request.args)
+
+        node_id = request.args.get("node_id", type=str)
+
         if not node_id:
-            logger.debug("Registration failed: id missing")
-            return f"Error: id missing\n", 500
+            return "error: node id must be provided.\n", HTTPStatus.BAD_REQUEST
 
-        logger.debug("Register user [{}]".format(node_id))
+        if not valid_node_id(node_id):
+            return "error: invalid node_id. (node ids must be between 6 and 32 [a-z0-9] characters.)\n", HTTPStatus.BAD_REQUEST
```

Comment:
i am (in general) not a big fan of the code logic and the log printing the error in that logic being so far apart.

you test for the `[A-Z0-9]{6,32}` (capital letters actually) in the `valid_node_id` function, but then print what the "passing conditions" are here.  it is prone to error.  In these cases, I vote to be less specific then potentially misleading.

Reply:
I agree. I'll update the error to be less specific.

(In fact, it turned out that check was wrong anyway. Turns out the database is only configured to handle up to 16 characters...)